redash/query_runner/pg.py

```diff
@@ -284,6 +349,27 @@ def configuration_schema(cls):
             "required": ["dbname", "user", "password", "host", "port"],
             "secret": ["password"],
         }
+        if redshift_role_enabled:
```

Comment:
Having all the options under one data source will make the configuration a bit confusing. Also it makes it hard to enforce required fields.

How about we create a subclass for the IAM based authentication?

Reply:
Sure - that's a sensible option. So we end up with something like `Redshift - User Login` and `Redshift - IAM Login`.

Reply:
I would go with `Redshift` and `Redshift (IAM Login)`.